meos/src/general/lifting.c

```diff
@@ -1560,13 +1560,9 @@ tfunc_temporal_temporal(const Temporal *temp1, const Temporal *temp2,
               (TInstant *) temp1, (TSequence *) temp2, lfinfo) :
             (Temporal *) tfunc_tinstant_tcontseq(
               (TInstant *) temp1, (TSequence *) temp2, lfinfo);
-        case TSEQUENCESET:
+        default: /* TSEQUENCESET */
```

Comment:
Don't use `default` for the last case.
Since our possible cases are fixed, simply put each in a separate `case ...` and don't use a `default` at all for this match case section.

Reply:
Alas, doing that will produce warnings that we would like to avoid. This is the reason of having an assert before the switch

Reply:
The issue comes from 2 aspects:
1. the `subtype` field is a `uint8` instead of the `tempSubType` enum type for performance reasons (since we know we don't need more than 1 byte for the subtype value).
2. `tempSubType` has a 4th value `ANYTEMPSUBTYPE`

My ideas:

For 1: cast `temp->subtype` to `tempSubType` in a match statement: `match ((tempSubType) temp->subtype) ...`. This will let the compiler know that we are matching over an enum.
If we cover all cases we will thus not have any warnings.

For 2, two possible solutions:
1. Add an additional `case ANYTEMPSUBTYPE` instead of `default`, containing a meos_error.
2. Remove `ANYTEMPSUBTYPE` from the enum values and add it as a `#define ANYTEMPSUBTYPE 0` right above the enum definition. This should not break any of the other code in analyze and selectivity using `ANYTEMPSUBTYPE` (I tested the compilation and there were no errors or warnings) and will allow us to switch over only the 3 values in `tempSubType`

I think solution 2 is better with regard to the `match case` solution, but I'd accept solution 1 too.

Reply:
We ended up using `default` for the `TSEQUENCESET` case, as this is similar to the previous `if else` statement syntax. The `assert` makes sure that this default only hits `TSEQUENCESET` cases.